var/spack/repos/builtin/packages/py-xarray/package.py

```diff
@@ -12,6 +12,10 @@ class PyXarray(PythonPackage):
     homepage = "https://github.com/pydata/xarray"
     url      = "https://pypi.io/packages/source/x/xarray/xarray-0.9.1.tar.gz"
 
+    version('0.14.0', sha256='a8b93e1b0af27fa7de199a2d36933f1f5acc9854783646b0f1b37fed9b4da091')
+    version('0.13.0', sha256='80e5746ffdebb96b997dba0430ff02d98028ef3828e6db6106cbbd6d62e32825')
+    version('0.12.0', sha256='856fd062c55208a248ac3784cac8d3524b355585387043efc92a4188eede57f3')
+    version('0.11.0', sha256='636964baccfca0e5d69220ac4ecb948d561addc76f47704064dcbe399e03a818')
     version('0.9.1', sha256='89772ed0e23f0e71c3fb8323746374999ecbe79c113e3fadc7ae6374e6dc0525')
 
     depends_on('py-setuptools',      type='build')
```

Comment:
Dependencies have been updated:
```python
PYTHON_REQUIRES = ">=3.6"                                                       
INSTALL_REQUIRES = ["numpy >= 1.14", "pandas >= 0.24"]                          
needs_pytest = {"pytest", "test", "ptr"}.intersection(sys.argv)                 
SETUP_REQUIRES = ["pytest-runner >= 4.2"] if needs_pytest else []               
TESTS_REQUIRE = ["pytest >= 2.7.1"]   
```
You'll have to find the last version that supports Python 2 so we don't break builds for Python 2 users.

Reply:
I traced the required versions of python and other dependencies. At first I set a version range of the dependencies so that the maximal version fits the minimal dependency of the next higher version (see f991e07), but if there is no API break this would be too strict. What is in this case the best practise? Open end (i.e. only minimal version) or version intervals with empty intersection?

Reply:
What you have now is perfect!